### PR TITLE
Proxy version command HTTP call

### PR DESCRIPTION
**Pull request by viveksyngh**
This commit proxies the HTTP calls made in `version` command.

Fixes : #519

Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
@ewilde please could you review?

**Comment by alexellis** [comment]
@viveksyngh thank you for working on this so quickly. Going forward, I would like to see all organisation members supplying details of testing. I'll start calling this out more when it's not been done.

> How Has This Been Tested?
(empty)

**Comment by burtonr** [comment]
Ran through a couple scenarios and found everything to be working as expected

Also ran `go test` to ensure the existing tests still pass :+1: 

* Happy path:
```
$ ./faas-cli version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  
 version: dev

Gateway
 uri:     http://127.0.0.1:8080
 version: dev
 sha:     7a2a3542da5d5f67af001e6cd9948672945a0495
 commit:  update to add event to community listings for 2018


Provider
 name:          faas-swarm
 orchestration: swarm
 version:       dev 
 sha:           be442da69feaeff3e19be1ecbc18c1c3b2ef1ff5
```

* Not logged in:
```
$ ./faas-cli version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  
 version: dev
Error while making request to /system/info: unauthorized access, run "faas-cli login" to setup authentication for this server
```

* Not an OpenFaaS gateway URL:
```
$ ./faas-cli version -g http://blah.com
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  
 version: dev
Error while making request to /system/info: cannot connect to OpenFaaS on URL: http://blah.com
```

File: proxy/version.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+//GetSystemInfo get system information from /system/info endpoint
+func GetSystemInfo(gateway string, tlsInsecure bool) (map[string]interface{}, error) {
+	infoEndPoint := gateway + "/system/info"
+	timeout := 5 * time.Second
+
+	client := MakeHTTPClient(&timeout, tlsInsecure)
+	req, err := http.NewRequest("GET", infoEndPoint, nil)
+	if err != nil {
+		return nil, fmt.Errorf("cannot connect to OpenFaaS on URL: %s", gateway)
```
Comment (ewilde) [inline proxy/version.go]: don't think this is a connection err, types of errors thrown from  `http.NewRequest` are invalid http method or invalid url

**Comment by viveksyngh** [comment]
@alexellis Sorry for not adding test results. I had done testing on the local system similar to what @burtonr  has added but I forgot to add that in PR. 

Thanks, @burtonr for adding test results :)

File: proxy/version.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	response, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("cannot connect to OpenFaaS on URL: %s", gateway)
```
[the remark this answers is not in the document]
Comment (viveksyngh) [reply proxy/version.go]: Output might look similar to this 
```
$ ./faas-cli version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:  
 version: dev
Error while making request to /system/info: unauthorized access, run "faas-cli login" to setup authentication for this server
```
Should we remove the print statement when the call to `/system/info` endpoint fails?

**Comment by viveksyngh** [comment]
Test results after changes.

```
./faas-cli version
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:
 version: dev

Gateway
 uri:     http://127.0.0.1:8080
 version: 0.9.0
 sha:     9c2f6dd2a4e2aac0b253b29cc031865b76802ed5
 commit:  Fix bug introduced by URL transformer


Provider
 name:          faas-swarm
 orchestration: swarm
 version:       0.4.1
 sha:           50e03508a2b120d9b088485787725719f792f832
➜  faas-cli git:(issue-519) ✗ ./faas-cli version -g http://test.com
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|

CLI:
 commit:
 version: dev
```

File: proxy/version.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexellis) [reply proxy/version.go]: Yes I think that makes sense. Thanks @ewilde for the input

[the remark this answers is not in the document]
Comment (alexellis) [reply proxy/version.go]: So to clarify - no error in this scenario. Keep it silent.

**Comment by alexellis** [comment]
Let's get the milestone set.

**Comment by martindekov** [comment]
Derek set milestone: 0.7.2

**Pull request merged**